pkg/util/k8sutil/tpr_util.go

```diff
@@ -74,16 +79,36 @@ func GetClusterTPRObject(restcli rest.Interface, ns, name string) (*spec.Cluster
 	return readOutCluster(b)
 }
 
-// UpdateClusterTPRObject updates the given TPR object.
-// ResourceVersion of the object MUST be set or update will fail.
-func UpdateClusterTPRObject(restcli rest.Interface, ns string, c *spec.Cluster) (*spec.Cluster, error) {
-	if len(c.Metadata.ResourceVersion) == 0 {
-		return nil, errors.New("k8sutil: resource version is not provided")
-	}
-	return updateClusterTPRObject(restcli, ns, c)
+// AtomicUpdateClusterTPRObject takes an update function which is executed before attempting
+// to update the Cluster TPR object. Upon conflict or some other error the entire process is retried
```

Comment:
> Upon conflict or some other error the entire process is retried

Let's retry only if there is a resource conflict. Any other error should return.